test_gen_js.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash -x
+
+# node/  javascript top-level bindings test 
+
+# we can now use '.tests.js' autogenerated javascript to build/pack one of each packet with test data for nominally sending
+
+# utilise the C bindings and test suite to generate a bunch of mavlink data that we can run through the node/javascript parser
+# we do this by executing ./generator/C/test/posix/testmav2.0_ardupilotmega to 
+# run mavlink_test_all() from generator/C/include_v2.0/ardupilotmega/testsuite.h
+
+set -e
+set -x
+
+test -z "$MDEF" && MDEF="../message_definitions"
+
+# build js bindings we want to test
+cd generator
+./gen_js.sh
+cd ..
+
+# run the automatically generated tool for build/pack, ie create and pack one of everything, no byte-level checking of the packed results tho, comes later.
+node generator/javascript/implementations/mavlink_ardupilotmega_v2.0/mavlink.tests.js
+node generator/javascript/implementations/mavlink_ardupilotmega_v1.0/mavlink.tests.js
+node generator/javascript/implementations/mavlink_common_v2.0/mavlink.tests.js
+node generator/javascript/implementations/mavlink_common_v1.0/mavlink.tests.js
+
+# build C bindings to test the testsuite.js 'recieve' tester against
+mavgen.py --lang C $MDEF/v1.0/ardupilotmega.xml -o generator/C/include_v1.0 --wire-protocol=1.0
+mavgen.py --lang C $MDEF/v1.0/ardupilotmega.xml -o generator/C/include_v2.0 --wire-protocol=2.0
+
+
+# pipe the wire-ready hex-output from the C bindings into the node bindings per-packet and recieve each of them
+pushd generator/C/test/posix
+make clean testmav1.0_ardupilotmega testmav2.0_ardupilotmega testmav1.0_common testmav2.0_common
+./testmav1.0_ardupilotmega | grep '^fe' | node ../../../../examples/testparser.js ardupilotmega 1.0
+./testmav2.0_ardupilotmega | grep '^fd' | node ../../../../examples/testparser.js ardupilotmega 2.0
+./testmav1.0_common | grep '^fe' | node ../../../../examples/testparser.js common 1.0
+./testmav2.0_common | grep '^fd' | node ../../../../examples/testparser.js common 2.0
+popd
+
+
+# make big collection ~990 of mocha tests based on C output like the above but more thorough, this includes byte-level checking of the packed results
```

Comment:
This section can be simplified to:
```
cd generator/javascript/ 
npm test
cd ../..
```
Also avoids the need to have mocha globally installed.